src/dali/plugin/input/rest/kraken.py

```diff
@@ -226,174 +196,150 @@ def _compute_tx_set(self, trade_history, ledger, processed_transactions) -> List
-                        'from_holder': self.account_holder if is_withdrawal else Keyword.UNKNOWN.value,
-                        'to_exchange': self.__EXCHANGE_NAME if is_deposit else Keyword.UNKNOWN.value,
-                        'to_holder': self.account_holder if is_deposit else Keyword.UNKNOWN.value,
-                        'crypto_sent': Keyword.UNKNOWN.value if is_deposit else str(amount),
-                        'crypto_received': str(amount) if is_deposit else Keyword.UNKNOWN.value,
-                    }
-                )
                 result.append(
                     IntraTransaction(
-                        plugin=kwargs[Keyword.PLUGIN.value],
-                        unique_id=kwargs[Keyword.UNIQUE_ID.value],
-                        raw_data=kwargs[Keyword.RAW_DATA.value],
-                        timestamp=kwargs[Keyword.TIMESTAMP.value],
-                        asset=kwargs[Keyword.ASSET.value],
-                        from_exchange=kwargs[Keyword.FROM_EXCHANGE.value],
-                        from_holder=kwargs[Keyword.FROM_HOLDER.value],
-                        to_exchange=kwargs[Keyword.TO_EXCHANGE.value],
-                        to_holder=kwargs[Keyword.TO_HOLDER.value],
-                        spot_price=kwargs[Keyword.SPOT_PRICE.value],
-                        crypto_sent=kwargs[Keyword.CRYPTO_SENT.value],
-                        crypto_received=kwargs[Keyword.CRYPTO_RECEIVED.value],
-                        notes=kwargs[Keyword.NOTES.value],
+                        plugin=self.__PLUGIN_NAME,
+                        unique_id=Keyword.UNKNOWN.value,
+                        raw_data=raw_data,
+                        timestamp=timestamp_value,
+                        asset=asset,
+                        from_exchange=self.__EXCHANGE_NAME if is_withdrawal else Keyword.UNKNOWN.value,
+                        from_holder=self.account_holder if is_withdrawal else Keyword.UNKNOWN.value,
+                        to_exchange=self.__EXCHANGE_NAME if is_deposit else Keyword.UNKNOWN.value,
+                        to_holder=self.account_holder if is_deposit else Keyword.UNKNOWN.value,
+                        spot_price=spot_price,
+                        crypto_sent=Keyword.UNKNOWN.value if is_deposit else str(amount),
+                        crypto_received=str(amount) if is_deposit else Keyword.UNKNOWN.value,
+                        notes=key,
                     )
                 )
                 continue
 
-            kwargs.update({
-                'exchange': self.__EXCHANGE_NAME,
-                'holder': self.account_holder,
-                'transaction_type': Keyword.BUY.value if RP2Decimal(record[_AMOUNT]) > ZERO else Keyword.SELL.value,
-                'crypto_fee': '0' if is_fiat_asset else record[_FEE],
-                'fiat_fee': record[_FEE] if is_fiat_asset else None,
-            })
+            crypto_fee: str = '0' if is_fiat_asset else record[_FEE]
```

Comment:
You need to have `crypto_fee` or `fiat_fee` not both, not even 0. It's probably best to change this to:
```
crypto_fee: str = None if is_fiat_asset else record[_FEE]
```

Reply:
It should only retrieve prices for transactions that have `Keyword.UNKNOWN.value` for their `spot_price` unique ids have no relation.

Kraken restricts it's REST API to one call _every 5 seconds_, so if you have a lot of transactions pulling data can take a very long time. It takes me about a day or so. If you have a valid Google Drive API key, it will pull OHLCVT CSV files from a folder, chunk and save them locally. That will speed things up a lot. However, they do not include the current quarter's worth of data, so that has to be pulled via the REST API.

Also, you can terminate the process (CTRL + C on Linux) and that will save all pricing info to the cache, so that you can resume from where you left off. Just be aware that running `pytest` will empty the cache, so you should back it up before you do that if you want to save your progress. Or, you can have two different installs of Dali-RP2 in two different folders.

Reply:
> 1). So it does cache, I see the file now in .dali_cache. But some of the spot_prices from the pair converter is incorrect. For example, prices that could not be possible was being set. Think post-20k prices being set in places where sub-20k prices could only be possible.

Are the timestamps accurate on the ods that is generated? That would be my first step in troubleshooting.

Reply:
I'll check. I was using the ccxt conversion functions so it could be my inputs. Maybe I'll roll back the function and try again to see if I did it wrong.

Reply:
So the reason for the erroneous spot_prices is because some of the asset symbols that Kraken uses are their own. For example, BTC is XBT so when transaction.asset is set to BTC, the pair_converter looks for BTC_OHLCVT.zip which doesn't exist causing all sorts of issues. It will want to download XBT_OHLCVT.zip. I'm currently trying to figure out a way to fix this.

Reply:
@macanudo527 I forget why does the CCXT pair plugin works without google API key? One with google API key and one without google API key.

Reply:
The CCXT pair converter needs the Google API key to download OHLCVT information. ([more information](https://github.com/eprbell/dali-rp2/blob/main/docs/configuration_file.md#ccxt))